solutions/kubernetes/ceph-binary-build.sh

```diff
@@ -0,0 +1,299 @@
+
+    add_common_separator "List generated binary packages (*.deb)"
+    ls *.deb
+popd
+EOF
+
+    chmod +x build.sh
+    popd
+
+    add_secondary_separator "Run build script"
+    docker exec ceph_ubuntu bash /home/build.sh
+
+    add_primary_separator "\t\tDestroy build environment"
+    docker stop ceph_ubuntu
+}
+
+function build_centos() {
+    add_primary_separator "Building centos binary packages"
+
+    add_secondary_separator "Verfiy/pull CentOS docker image"
+    if [[ $(docker images | tail -n+2 | awk '{ print $1":"$2 }' | grep centos) != "centos:latest" ]]; then
+        docker pull centos:latest
+    fi
+
+    add_secondary_separator "Make build directory and run CentOS container"
+    mkdir -p $MOUNT_LOCATION/centos
+    docker run --rm -t -d --name ceph_centos -v /$MOUNT_LOCATION/centos:/home centos:latest
+
+    pushd $MOUNT_LOCATION/centos
+        add_common_separator "Removing previous files"
+        rm -rvf *
+    popd
+
+    add_secondary_separator "Write build script"
+    cp functions.sh $MOUNT_LOCATION/centos
+    pushd $MOUNT_LOCATION/centos
+
+        cat << EOF > build.sh
+#/bin/bash
+
+source /home/functions.sh
+
+add_common_separator "Update repolist cache and install prerequisites"
+rpm -ivh http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/Packages/centos-gpg-keys-8-3.el8.noarch.rpm
+dnf --disablerepo '*' --enablerepo=extras swap centos-linux-repos centos-stream-repos -y
+yum makecache && yum install git -y
+yum install wget bzip2 rpm-build rpmdevtools dnf-plugins-core -y
+dnf config-manager --set-enabled powertools
+pushd /home/
+    add_common_separator "Clone Repo"
+    git clone $CEPH_REPO -b $CEPH_BRANCH
+
+    pushd ceph
+        add_common_separator "Checkout Submodules"
+        git submodule update --init --recursive
+
+        add_common_separator "Install Dependencies"
+        ./install-deps.sh
+
+        add_common_separator "Make Source Tarball"
+        ./make-dist
+        
+        mkdir -p ../rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
+        tar --strip-components=1 -C ../rpmbuild/SPECS --no-anchored -xvjf ceph-*tar.bz2 "ceph.spec"
+        mv ceph*tar.bz2 ../rpmbuild/SOURCES/
+    popd
+
+    pushd rpmbuild/
+        add_common_separator "Start Build"
+        rpmbuild --define "_topdir /home/rpmbuild" -ba SPECS/ceph.spec
+    popd
+
+    add_common_separator "List generated binary packages (*.rpm)"
+    ls rpmbuild/RPMS/*
+popd
+EOF
+
+    chmod +x build.sh
+    popd
+
+    add_secondary_separator "Run build script"
+    docker exec ceph_centos bash /home/build.sh
+
+    add_primary_separator "\t\tDestroy build environment"
+    docker stop ceph_centos
+}
+
+function build_rockylinux() {
+    add_primary_separator "Building rocky linux binary packages"
+
+    add_secondary_separator "Verfiy/pull Rocky Linux docker image"
+    if [[ $(docker images | tail -n+2 | awk '{ print $1":"$2 }' | grep rockylinux) != "rockylinux:latest" ]]; then
+        docker pull rockylinux:latest
```

Comment:
Let's use some fixed version instead of the latest. Latest is a kind of moving target and may break.